py-polars/polars/series/series.py

```diff
@@ -2986,8 +2986,17 @@ def filter(self, predicate: Series | list[bool]) -> Self:
                 1
                 3
         ]
+        >>> s.filter(pl.element() != 2)
+        shape: (2,)
+        Series: 'a' [i64]
+        [
+                1
+                3
+        ]
         """
-        if isinstance(predicate, list):
+        if isinstance(predicate, pl.Expr):
+            return self.to_frame("").filter(predicate).to_series().rename(self.name)
+        elif isinstance(predicate, list):
             predicate = Series("", predicate)
         return self._from_pyseries(self._s.filter(predicate._s))
```

Comment:
How about just do
```python
return self.to_frame("").filter(predicate).to_series().rename(self.name)
```
It should just work without the extra type checks.

Reply:
Hmm, maybe good suggestion. That introduces a bit of overhead but would allow us to remove the Series binding. Perhaps I will do this--thanks!